extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -187,9 +249,39 @@ export class AmpAdExit extends AMP.BaseElement {
       this.transport_.beacon = config.transport[TransportMode.BEACON] !== false;
       this.transport_.image = config.transport[TransportMode.IMAGE] !== false;
     } catch (e) {
-      this.user().error(TAG, 'Invalid JSON config', e);
+      user().error(TAG, 'Invalid JSON config', e);
       throw e;
     }
+
+    this.unlisten_ = listen(this.getAmpDoc().win, 'message', event => {
+      const responseMessage = deserializeMessage(getData(event));
+
+      let ampAdResourceId;
+      try {
+        ampAdResourceId = this.element.ownerDocument.defaultView
```

Comment:
cache the ampAdResourceId in buildCallback?

Reply:
Fixed.
Agree re: splitting, will do that tomorrow.